src/main/java/com/minecolonies/core/colony/ColonyManager.java

```diff
@@ -185,31 +198,44 @@ private void deleteColony(@Nullable final IColony iColony, final boolean canDest
 
             try
             {
+                Log.getLogger().info("Unregistering event handlers for {}", id);
                 MinecraftForge.EVENT_BUS.unregister(colony.getEventHandler());
             }
             catch (final NullPointerException e)
             {
                 Log.getLogger().warn("Can't unregister the event handler twice");
             }
 
-            Log.getLogger().info("Deleting colony: " + colony.getID());
+            // Delete the colony
+            Log.getLogger().info("Deleting colony {}", colony.getID());
+            cap.deleteColony(id);
 
-            final IColonyManagerCapability cap = world.getCapability(COLONY_MANAGER_CAP, null).resolve().orElse(null);
-            if (cap == null)
+            // Delete the colony team
+            Log.getLogger().info("Deleting team for colony {}", id);
+            final PlayerTeam team = colony.getTeam();
+            if (team != null)
             {
-                Log.getLogger().warn(MISSING_WORLD_CAP_MESSAGE);
-                return;
+                world.getScoreboard().removePlayerTeam(team);
+            }
+
+            // Send the deletion event
+            try
+            {
+                MinecraftForge.EVENT_BUS.post(new ColonyDeletedEvent(colony));
```

Comment:
we probably should have a util doing this for all events, so we can just do ColonyEvents.send(event)

Reply:
@Thodor12